stan/math/prim/mat/prob/neg_binomial_2_log_glm_lpmf.hpp

```diff
@@ -133,8 +129,8 @@ neg_binomial_2_log_glm_lpmf(const T_y& y, const T_x& x, const T_alpha& alpha,
   T_precision_val log_phi = log(phi_arr);
   Array<T_partials_return, Dynamic, 1> logsumexp_theta_logphi
       = (theta > log_phi)
-            .select(theta + log1p(exp(log_phi - theta)),
-                    log_phi + log1p(exp(theta - log_phi)));
+            .select(theta + Eigen::log1p(Eigen::exp(log_phi - theta)),
+                    log_phi + Eigen::log1p(Eigen::exp(theta - log_phi)));
```

Comment:
With all the autos, I can't tell the types here, but if these are autodiff types, we want to be applying our own functions rather than Eigen's.  

Are these select functions performant with the comparison?  

And why roll your own log-sum-exp this way?  Can't you just use the vectorized one?